File: Backend/src/main/java/com/project/autoserve/service/DashboardService.java
package com.project.autoserve.service;

import com.project.autoserve.dto.dashboard.AdminDashboardResponseDTO;
import com.project.autoserve.dto.dashboard.CustomerDashboardResponseDTO;
import com.project.autoserve.dto.dashboard.MechanicDashboardResponseDTO;

public interface DashboardService {

    AdminDashboardResponseDTO getAdminDashboard();

    CustomerDashboardResponseDTO getCustomerDashboard();

    MechanicDashboardResponseDTO getMechanicDashboard();

}

File: Frontend/src/services/dashboardService.js
import axiosInstance from '../api/axiosInstance';

export const dashboardService = {
  getAdminDashboard: async () => {
    const response = await axiosInstance.get('/dashboard/admin');
    return response.data;
  },

  getCustomerDashboard: async () => {
    const response = await axiosInstance.get('/dashboard/customer');
    return response.data;
  },

  getMechanicDashboard: async () => {
    const response = await axiosInstance.get('/dashboard/mechanic');
    return response.data;
  },
};
